akka-http-testkit/src/main/scala/akka/http/javadsl/testkit/TestRoute.scala

```diff
@@ -17,5 +17,16 @@ import akka.http.javadsl.server.Route
  */
 trait TestRoute {
   def underlying: Route
+
+  /**
+   * Run the request against the sealed route, meaning use a default handler
+   * as a fallback for all cases the route doesn’t handle itself.
+   */
   def run(request: HttpRequest): TestRouteResult
-}
\ No newline at end of file
+
+  /**
+   * Run the request against the semi-sealed route, meaning that a default
+   * handler is used for exceptions but not for rejections.
```

Comment:
A longer form of that documentation:

```
Run the request against the "semi-sealed" route, meaning that exceptions will be handled by the default exception handler but rejections will not be handled. This means that the test will not see HTTP responses with error status codes for routes that rejected a request. Instead, the `TestRouteResult` allows access to the original rejection containing all the rejection details in structured form. Use `assertRejections` to check that a route rejected a request with expected rejections.

Otherwise, to assert on the actual error HTTP response generated by the default rejection handler, use the `run` method, instead.
```